src/repos/UserRepo.ts

```diff
@@ -75,8 +87,49 @@ const getUsersFollowingPublication = async (pubSlug: string): Promise<User[]> =>
   return users;
 };
 
+/**
+ * Add article to a user's readArticles
+ */
+const appendReadArticle = async (uuid: string, articleID: string): Promise<User> => {
+  const article = await ArticleRepo.getArticleByID(articleID);
+  const user = await UserModel.findOne({ uuid });
+  const checkDuplicates = (prev: boolean, cur: Article) => prev || cur.id === articleID;
+  if (!user.readArticles.reduce(checkDuplicates, false)) {
+    user.readArticles.push(article);
+  }
+  user.save();
+  return user;
+};
+
+/**
+ * Increment shoutouts in user's numShoutouts
+ */
+const incrementShoutouts = async (uuid: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+  if (user !== null) {
+    user.numShoutouts += 1;
+    user.save();
+  }
+  return user;
```

Comment:
if the user is null, doesn't this show up as a graphql error? or is it just an empty object?

Reply:
Fixed! Should return null now in that case.